### PR TITLE
Remove pointer structures that don't require to be pointers.

### PR DESCRIPTION
**Pull request by calavera**
Signed-off-by: David Calavera david.calavera@gmail.com

### CODE REVIEW

File: daemon/inspect.go [hunk 1]
```diff
@@ -32,7 +32,7 @@ func (daemon *Daemon) ContainerInspect(name string) (*types.ContainerJSON, error
 		})
 	}
 
-	return &types.ContainerJSON{base, mountPoints, container.Config}, nil
+	return &types.ContainerJSON{*base, mountPoints, *container.Config}, nil
```
Comment (LK4D4) [inline daemon/inspect.go]: I don't like asterisks all over in code. It complicates reading code.

File: api/types/types.go [hunk 2]
```diff
@@ -240,26 +240,26 @@ type ContainerJSONBase struct {
...
-	HostConfig      *runconfig.HostConfig
+	HostConfig      runconfig.HostConfig
 	GraphDriver     GraphDriverData
 }
 
 type ContainerJSON struct {
-	*ContainerJSONBase
+	ContainerJSONBase
 	Mounts []MountPoint
-	Config *runconfig.Config
```
Comment (LK4D4) [inline api/types/types.go]: Config is *VERY* big structure with maps and slices. This should be pointer according to Google style guide https://github.com/golang/go/wiki/CodeReviewComments

File: daemon/inspect.go (hunk 1 above)
Comment (calavera) [inline daemon/inspect.go]: it's not like this is the only code where asterisks exist :wink:

I prefer to use values for embedded structs than pointers. It's safer to call methods from the parent this way.

Comment (LK4D4) [inline daemon/inspect.go]: It hurts readability. Safety isn't nearly issue in our case.

**Comment by LK4D4** [comment]
I personally don't like changes. It potentially can create a lot of garbage because of copying large complicated structures. Also such code(where you copy structures for safety) isn't common in golang world.

**Comment by calavera** [comment]
you're right. Closing then, thanks for the review.

**Pull request closed without merging**